pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotApplicationQuotaRestletResource.java

```diff
@@ -0,0 +1,139 @@
+@SwaggerDefinition(securityDefinition = @SecurityDefinition(apiKeyAuthDefinitions = {
+    @ApiKeyAuthDefinition(name = HttpHeaders.AUTHORIZATION, in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key =
+        SWAGGER_AUTHORIZATION_KEY, description =
+        "The format of the key is  ```\"Basic <token>\" or \"Bearer "
+            + "<token>\"```"), @ApiKeyAuthDefinition(name = CommonConstants.APPLICATION, in =
+    ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key = CommonConstants.APPLICATION, description =
+    "Application context passed through http header. If no context is provided 'default' application "
+        + "context will be considered.")
+}))
+@Path("/")
+public class PinotApplicationQuotaRestletResource {
+  public static final Logger LOGGER = LoggerFactory.getLogger(PinotApplicationQuotaRestletResource.class);
+
+  @Inject
+  PinotHelixResourceManager _pinotHelixResourceManager;
+
+  /**
+   * API to get application quota configs. Will return null if application quotas are not defined
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/applicationQuotas")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_APPLICATION_QUERY_QUOTA)
+  @ApiOperation(value = "Get all application qps quotas", notes = "Get all application qps quotas")
+  public Map<String, Double> getApplicationQuotas(@Context HttpHeaders httpHeaders) {
+    Map<String, Double> quotas = _pinotHelixResourceManager.getApplicationQuotas();
+    if (quotas != null) {
+      return quotas;
+    } else {
+      return Collections.emptyMap();
+    }
+  }
+
+  /**
+   * API to get application quota configs. Will return null if application quotas are not defined
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/applicationQuotas/{appName}")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_APPLICATION_QUERY_QUOTA)
+  @ApiOperation(value = "Get application qps quota", notes = "Get application qps quota")
+  public Double getApplicationQuota(@Context HttpHeaders httpHeaders, @PathParam("appName") String appName) {
+
+    Map<String, Double> quotas = _pinotHelixResourceManager.getApplicationQuotas();
+    if (quotas != null && quotas.containsKey(appName)) {
+      return quotas.get(appName);
+    }
+
+    HelixConfigScope scope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).forCluster(
+        _pinotHelixResourceManager.getHelixClusterName()).build();
+    HelixAdmin helixAdmin = _pinotHelixResourceManager.getHelixAdmin();
+    String defaultQuota =
+        helixAdmin.getConfig(scope, Collections.singletonList(CommonConstants.Helix.APPLICATION_MAX_QUERIES_PER_SECOND))
+            .getOrDefault(CommonConstants.Helix.APPLICATION_MAX_QUERIES_PER_SECOND, null);
+    return defaultQuota != null ? Double.parseDouble(defaultQuota) : null;
+  }
+
+  /**
+   * API to update the quota configs for application
+   */
+  @POST
```

Comment:
Yes, it mimicks db quota rest api. I can add DELETE if needed, no problem.

Reply:
I thought the delete API would be more confusing as even if you delete the query quota, the default query quota (if configured) will come into effect so unsetting the override (via POST) seemed the right operation rather than delete. But we can revisit this as now I am thinking POST is also confusing and it should have been a separate PUT...